### ISSUE TITLE
Implement Azure AD 

### ISSUE THREAD

**Issue by allanhaggett** [allanhaggett/learning-curator#242]
JSON Web Tokens moves session management onto the client and is preferable in stateless apps.

**Comment by allanhaggett** [comment]
CakePHP framework supports JWT natively :) super-duper simplifying things!

https://book.cakephp.org/authentication/2/en/authenticators.html#jwt

**Comment by allanhaggett** [comment]
Huzzah! I'm not gonna lie: I got a bit emotional when I got this working yesterday. It's not captured very well here, but closing this issue is actually the answer to a big, open question that's been hanging over this project since its inception, and is probably the biggest issue closed to date! 

Note that this still needs to be fully tested and deployed to OpenShift. The code that makes it happen currently only resides as  uncommitted code on my virtual server (_not deployed to silver cluster_). It needs to basically stay this way until we figure out where we store sensitive secret information e.g. API keys to AzureAD (so this closure isn't tied to a specific commit). There are also database schema updates and a variety of other stuff that needs to be done before this is actually fully "implemented" and in production, but getting to where we are here is a **huge** milestone and so I'm closing this particular issue to reflect that. We'll break out next steps into issues at our planning session on Thursday this week.

Thanks to Amir (and Jared) via Freshworks for getting me unstuck and validating my approach, and both Warren and @Kousthubha-M for their contributions in those meetings with Amir and beyond :)
